governator-api/src/main/java/com/netflix/governator/event/ApplicationEventDispatcher.java

```diff
@@ -0,0 +1,19 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+
+/**
+ * Interface to publishing {@link ApplicationEvent}s as well as programmatically registering
+ * {@link ApplicationEventListener}s.
+ */
+public interface ApplicationEventDispatcher {
+
+    <T extends ApplicationEvent> void registerListener(Class<T> eventType, ApplicationEventListener<T> eventListener);
```

Comment:
Should we consider adding an unregisterListener?  Or return a subscription on which unsubscribe() can be called.